examples/gitops/argocd-multi-cluster/README.md

```diff
@@ -0,0 +1,326 @@
+
+
+### UserPool configuration
+
+* You have a Cognito UserPool created (if not yet, make one right now, stick to the defaults if you do it through AWS Console). Note the `Pool Id`. We are going to use it later.
+* Create an app client per the correspondent Argo application.
+* Configure an app client per the correspondent Argo application.
+  * In `Enabled Identity Providers` select `Cognito User Pool`
+  * In `Callback URL(s)` specify `https://${your-ARGOCD-fqdn}/auth/callback`
+  * In `Sign out URL(s)` specify `https://${your-ARGOCD-fqdn}/logout`
+  * In `Allowed OAuth Flows` select `Authorization code grant`
+  * In `Allowed OAuth Scopes` select `email`, `openid`, `profile`
+  * Save Changes
+* Configure the domain name. It would be best if you had it for the authentication flow to work.
+  * For this guide, use `Amazon Cognito domain` with a custom `Domain prefix`, and set any value there. Cognito will use it to redirect you to your UserPoll Sign-In page. It is OK to use your domain as well, but its config is out of the scope of this guide. Take note of the field.
+
+### CLI authentication
+
+We need another secret-less **client application** in the Cognito user pool. Go and create one in AWS Console. Just make sure `Generate client secret` option is not selected. Please take note of it's client ID.
+Now configure this client app. Let's make sure the following is selected:
+* In `Enabled Identity Providers` select `Cognito User Pool`
+* In `Callback URL(s)` specify `http://localhost:8085/auth/callback`
+* In `Allowed OAuth Flows` select `Authorization code grant`
+* In `Allowed OAuth Scopes` select `email`, `openid`, `profile`
+* Save Changes
+
+### RBAC
+
+Add a user to the user pool, then create groups `argocd-admin` and `argocd-readonly` and add the users to each corresponding group. Add the user you just added to the `argocd-admin` to allow admin tasks in ArgoCD.
+
+The group names can be change in [./hub-cluster/cognito.yaml](./hub-cluster/cognito.yaml)
+
+```yaml
+configs:
+  rbac:
+    policy.csv: |
+      g, argocd-admin, role:admin
+      g, argocd-readonly, role:readonly
+    scopes: '[cognito:groups]'
+```
+ArgoCD built-in roles are `role:admin` and `role:readonly` additional policies can be added see [ArgoCD RBAC documentation](https://argo-cd.readthedocs.io/en/stable/operator-manual/rbac/)
+
+### CLI Login
+Log in with the Argo CLI using the command  `argocd login https://myargocd.acme.com --sso`.
+
+### Disable Dex
+ArgoCD also supports SSO using [Bundled Dex OIDC provider](https://argo-cd.readthedocs.io/en/stable/operator-manual/user-management/#dex)
+Since we are not using dex, the example disables dex in ArgoCD.
+
+
+### Deploy Hub Cluster with ArgoCD SSO
+
+Collect the following information from Amazon Cognito User Pool:
```

Comment:
[Future improvement] A lot of this information could be pulled from a Cognito data sources https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/cognito_user_pool_client which would reduce the amount of info that needs to be collected and injected via environment variables

Reply:
Good idea, I want to explore if Cognito can be created and setup in terraform instead of manually.